### PR TITLE
webhooks: add webhooks table

### PR DESCRIPTION
**Pull request by sashaostrikov**
Closes https://github.com/sourcegraph/sourcegraph/issues/42572

## Test plan
- sg migrate up -> downto -> up
- CI

### CODE REVIEW

File: internal/database/schema.md [hunk 1]
```diff
@@ -3044,6 +3044,27 @@ Foreign-key constraints:
...
+ code_host_urn  | text                     |           |          | 
+ secret         | text                     |           |          | 
+ created_at     | timestamp with time zone |           | not null | now()
+ updated_at     | timestamp with time zone |           | not null | now()
+
+```
+
+Webhooks registered in Sourcegraph instance.
+
+**code_host_kind**: Kind of an external service which webhooks are registered.
```
Comment (ryanslade) [inline internal/database/schema.md:3062]:
```suggestion
**code_host_kind**: Kind of an external service for which webhooks are registered.
```

File: internal/database/schema.md [hunk 2]
```diff
@@ -3044,6 +3044,27 @@ Foreign-key constraints:
 
 ```
 
+# Table "public.webhooks"
+```
+     Column     |           Type           | Collation | Nullable | Default 
+----------------+--------------------------+-----------+----------+---------
+ id             | uuid                     |           | not null | 
+ code_host_kind | text                     |           | not null | 
+ code_host_urn  | text                     |           |          |
```
Comment (ryanslade) [inline internal/database/schema.md:3053]: Maybe this should also be `not null`?

Comment (ryanslade) [reply internal/database/schema.md:3053]: Although I see it's also nullable in the repo table

**Review by ryanslade — APPROVED** [review]
LGTM

Just wondering about whether we want to the urn column to be nullable. Maybe this is a case where we can start strict and relax it alter if we find a good reason?

**Comment by sashaostrikov** [comment]
> LGTM
> 
> 
> 
> Just wondering about whether we want to the urn column to be nullable. Maybe this is a case where we can start strict and relax it alter if we find a good reason?

I agree, let's start strict in this case 👍

**Comment by sashaostrikov** [comment]
> I think we may also need an `encyption_key_id` `text` column. It looks like the other tables that support encryption have this.

found it. I'll make it not null with a default of an empty string as in other tables

**Comment by sashaostrikov** [comment]
@ryanslade PTAL, I've fixed the comments and added new column

**Review by ryanslade — APPROVED** [review]
Noice

**Pull request merged**